src/app/pet/pet-favourite/pet-favourite.component.ts

```diff
@@ -131,7 +130,17 @@ export class PetFavouriteComponent implements OnInit {
             }
         });
         console.log("Updated favourite pets for user "+this.user.username+".");
+        console.log(this.user.username+" now has "+this.user.favouritedPets.length+" pets marked as favourite!");
+        // For testing purposes, sleep and then reload to see console logs: 
+        //this.sleep(5000).then(()=>{window.location.reload();})
+        // Update page so findById recharges
+        window.location.reload();
```

Comment:
Is this reload really needed? It completely breaks Angular framework behaviour. If previous changes involve variables that are displayed through the template, front end updates should happen automatically...

Reply:
From what i've been testing, it's the only way i've found to "re-fetch" the data for the "findByUserIdAndPetId" method to work well with the data on the backend. 
Without this reload i can't manage to delete the correct entity since it keeps finding entities that have been deleted already.
If you have any idea on what could fix this i could look into it before the sprint ends